Sources/MongoSwift/MongoCursor.swift

```diff
@@ -41,6 +44,17 @@ public class MongoCursor<T: Codable>: Cursor {
         case cached(T?)
         /// Indicates that there is no value cached.
         case none
+
+        /// Get the contents of the cache and clear it.
+        fileprivate mutating func clear() -> T?? {
```

Comment:
The outer optional refers to whether or not the cache had anything it in, the inner one refers to whether the result returned from the server was nil or not.

Reply:
the API feels non-intuitive to me. did you consider just returning the previous value of `CachedDocument` and switching on that where this method is called?
I think this docstring should at least clarify what you've said in this comment.